Source/SelfService/Web/microservice/microserviceView/configuration/setupSection.tsx

```diff
@@ -0,0 +1,268 @@
+            ml: 0
+        }
+    },
+    formSections: {
+        mb: 4,
+        display: 'flex',
+        flexDirection: 'column'
+    }
+};
+
+const runtimeVersions: { value: string }[] = [];
+
+type SetupSectionProps = {
+    application: HttpResponseApplication;
+    applicationId: string;
+    environment: string;
+    microserviceId: string;
+    currentMicroservice: MicroserviceStore;
+};
+
+export type SetupSectionParameters = {
+    microserviceName: string;
+    developmentEnvironment: string;
+    imageName: string;
+    port: number;
+    entrypoint: string;
+    ingressPath: string;
+};
+
+export const SetupSection = ({ application, applicationId, environment, microserviceId, currentMicroservice }: SetupSectionProps) => {
+    const { enqueueSnackbar } = useSnackbar();
+    const history = useHistory();
+
+    const environmentInfo = application.environments.find(_environment => _environment.name === environment)!;
+    const canDelete = canDeleteMicroservice(application.environments, environment, microserviceId);
+
+    const { edit: { extra: {
+        headCommand,
+        runtimeImage,
+        headImage,
+        ingress
+    } } } = currentMicroservice;
+
+    const [setupPanelExpanded, setSetupPanelExpanded] = useState(true);
+    const [deleteDialogIsOpen, setDeleteDialogIsOpen] = useState(false);
+    const [formIsNotEditable, setFormIsNotEditable] = useState(true);
+    const [hasPublicURL, setHasPublicURL] = useState(false);
+    const [selectedRuntimeImage, setSelectedRuntimeImage] = useState('');
+    const [hasM3Connector, setHasM3Connetcor] = useState(environmentInfo.connections.m3Connector || false);
+    const [headCommandArgs, setHeadCommandArgs] = useState<string[]>(headCommand?.args || []);
+
+    useEffect(() => {
+        getRuntimes().forEach(runtime => {
+            runtimeVersions.push({ value: runtime.image.replace(/dolittle\/runtime:/gi, '') });
+        });
+        runtimeVersions.push({ value: 'None' });
+
+        setSelectedRuntimeImage(runtimeImage?.replace(/dolittle\/runtime:/gi, '') || 'None');
+    }, []);
+
+    const handleRestart = async () => {
+        await microserviceRestart({ applicationId, environment, microserviceId, enqueueSnackbar });
+    };
+
+    const handleDelete = async () => {
+        // Add loading state?
+        if (!canDelete) {
+            enqueueSnackbar('Deleting microservice is disabled', { variant: 'error' });
+            return;
+        }
+
+        const success = await deleteMicroservice(applicationId, environment, microserviceId);
+
+        if (!success) {
+            enqueueSnackbar('Failed to delete microservice', { variant: 'error' });
+            return;
+        }
+
+        enqueueSnackbar(`${currentMicroservice.name} has been deleted.`, { variant: 'info' });
+        const href = `/microservices/application/${applicationId}/${environment}/overview`;
+        history.push(href);
+    };
+
+    return (
+        <Box>
+            <AlertDialog
```

Comment:
Also outside the scope of this PR, but is `AlertDialog` the correct name for this component in the design system? Would something like `ConfirmationDialog` be better?

Also, the prop `handleDeletionConfirm` could be renamed to `handleConfirmation` or similar.